robot/impl/resource_manager_test.go

```diff
@@ -710,6 +710,8 @@ func TestManagerNewComponent(t *testing.T) {
 	robotForRemote := &localRobot{
 		manager: newResourceManager(resourceManagerOptions{}, logger),
 	}
+	// TODO: create an injectable dummy mod manager
+	robotForRemote.manager.moduleManager = &dummyModMan{}
```

Comment:
If you guard the `Add` call in `updateResources` with `if conf.Added.Modules != nil` I think you'll need neither this change nor the one below.